Patterns/LayeredProject.md

```diff
@@ -0,0 +1,100 @@
+# Layered project
+
+Keeping the structure of each .NET project somewhat similar helps developers to quickly find existing code and directly know where to add new code.
+
+## Solution layout
+
+A .NET solution must follow the following guidelines:
+
+- The name of the solution is `{Customer}.{Project}.sln`.
+- Put simple, solution-wide used objects like enums, exceptions and simple classes in Core: `{Customer}.{Project}.Core.csproj`.
+- Put everything that has to do with an external API of which the project is client of in a Gateway project: `{Customer}.{Project}.Gateway.{External API}.csproj`.
+- If the project exposes an API using ASP.NET Core Web Api, create a project for it called `{Customer}.{Project}.Api.csproj`.
+- If the project uses Azure Functions, create a project for those functions called `{Customer}.{Project}.Functions.csproj`.
+- Put all general business logic in `{Customer}.{Project}.Business.csproj`.
+- If the project is probably going to contain a lot of business logic, split the business logic up into parts and give each part a separate project: `{Customer}.{Project}.Business.Users.csproj`, `{Customer}.{Project}.Business.Payment.csproj`. Common business stuff can still be put in `{Customer}.{Project}.Business.csproj`.
+- If the project uses other (web) applications that do more than just an API, put them in a project that bests describes their functionality, like `{Customer}.{Project}.Cms.csproj`.
+- Put everything that has to do with the persistence layer in a separate project and use name that describes what it persists `{Customer}.{Project}.Data.csproj` (for all data if the project is not going to be very big) and, if needed, `{Customer}.{Project}.Data.User.csproj` (for just user related data).
+- Unit tests only test a specific project, and should always be called `{ProjectToTest}.Tests.csproj`.
+- Integration tests should be named like unit tests: `{ProjectToTest}.Tests.csproj`.
+
```

Comment:
- Always mirror the folder structure of the class under test. E.g. For the class '{Customer}.{Project}.Business.Comets.OrbitCalculations.cs' you should find it tests under '{Customer}.{Project}.Business.Tests.Comets.OrbitCalculationsTests.cs'

Reply:
Yes, I'll add something like that to this. I find myself sometimes adding folders for a specific object, because there are a lot of tests for it.

Reply:
Will be fixed.